src/Stryker.RegexMutators/Stryker.RegexMutators/RegexMutantOrchestrator.cs

```diff
@@ -0,0 +1,90 @@
+﻿using RegexParser;
+using RegexParser.Exceptions;
+using RegexParser.Nodes;
+using RegexParser.Nodes.AnchorNodes;
+using RegexParser.Nodes.CharacterClass;
+using RegexParser.Nodes.QuantifierNodes;
+using Stryker.RegexMutators.Mutators;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Stryker.RegexMutators
+{
+    public class RegexMutantOrchestrator
+    {
+        private readonly string _pattern;
+        private RegexNode _root;
+        private IDictionary<Type, IEnumerable<IRegexMutator>> _mutatorsByRegexNodeType;
+
+        public RegexMutantOrchestrator(string pattern)
+        {
+            _pattern = pattern;
+        }
+
+        public IEnumerable<RegexMutation> Mutate()
+        {
+            var parser = new Parser(_pattern);
+            try
+            {
+                RegexTree tree = parser.Parse();
+                _root = tree.Root;
+            }
+            catch (RegexParseException)
+            {
+                yield break;
+            }
+
+            _mutatorsByRegexNodeType = new Dictionary<Type, IEnumerable<IRegexMutator>>
+            {
+                {
+                    typeof(AnchorNode),
+                    new List<IRegexMutator>
+                    {
+                        new AnchorRemovalMutator(_root)
+                    }
+                },
+                {
+                    typeof(QuantifierNode),
+                    new List<IRegexMutator>
+                    {
+                        new QuantifierRemovalMutator(_root)
+                    }
+                },
+                {
+                    typeof(CharacterClassNode),
+                    new List<IRegexMutator>
+                    {
+                        new CharacterClassNegationMutator(_root)
+                    }
+                },
+                {
+                    typeof(CharacterClassShorthandNode),
+                    new List<IRegexMutator>
+                    {
+                        new CharacterClassShorthandNegationMutator(_root)
+                    }
+                },
+            };
```

Comment:
I would expect this to be set before the method is called. For example in the constructor. Also, the indentation is a bit much IMO. Maybe this could fit in fewer lines and improve readability